src/ModelerApp.vue

```diff
@@ -19,9 +19,11 @@
         </div>
       </b-card-header>
       <b-card-body class="overflow-hidden position-relative p-0 vh-100">
-        <modeler ref="modeler" @validate="validationErrors = $event" @warnings="warnings = $event" />
+        <modeler ref="modeler" @validate="validationErrors = $event" @warnings="warnings = $event" :decorations="decorations" />
       </b-card-body>
-      <validation-status :validation-errors="validationErrors" :warnings="warnings" />
+      <validation-status ref="validationStatus" :validation-errors="validationErrors" :warnings="warnings" :owner="self">
+        <component v-for="(component, index) in validationBar" :key="`validation-status-${index}`" :is="component" :owner="self" />
```

Comment:
I think `index` is sufficient for the `key` property: `:key="index"`.

I would highly recommend removing the `owner` prop. Is there a specific reason you did this instead of the regular props / events flow? I think it's a very hacky way to step around the Vue component API to enable two-way data binding, which is exactly what Vue's component based design aims to avoid as it's very hard to reason about and prone to bugs. The `owner` prop seems to be used how `$parent` would be used, but is more obscure.

In this case, I don't see any problems with passing down a `validationBar` prop, and listening for something like an `@update-validation-bar` event to update the `validationBar` data.

Reply:
removed